### ISSUE TITLE
Missing 'each' prefix in array modifier

### ISSUE THREAD

**Issue by casella** [open-ideas/IDEAS#1273]
There are several models in IDEAS that fail because of illegal use of scalar modifiers for arrays. Other tools accept that, but OMC doesn't, because it's not in the specification and can potentially lead to ambiguity

See, e.g. [IDEAS.Buildings.Components.Examples.BeamRadiationOnFloor](https://libraries.openmodelica.org/branches/master/IDEAS_dev/files/IDEAS_dev_IDEAS.Buildings.Components.Examples.BeamRadiationOnFloor.err) 
```
[/home/hudson/saved_omc/libraries/.openmodelica/libraries/IDEAS 3.0.0-master/Templates/Interfaces/BaseClasses/Structure.mo:50:3-52:73:writable]
Error: Non-array modification ‘"ThermodynamicTemperature"‘ for array component ‘quantity‘, possibly due to missing ‘each‘.
```
The offending code is [here](https://github.com/open-ideas/IDEAS/blob/b7a5c75d21a83ebde11f5a1abbb0a662e8963ec4/IDEAS/Templates/Interfaces/BaseClasses/Structure.mo#L50), it should read:
```modelica
Modelica.Blocks.Interfaces.RealOutput[nZones] TSensor(each final quantity="ThermodynamicTemperature",each unit="K", each displayUnit="degC", each min=0)
```
There are probably similar issues around, you can check the models in the [test report](https://libraries.openmodelica.org/branches/master/IDEAS/IDEAS.html) that fail in the frontend, but I guess they all refer to the same line of code, eventually.

**Comment by Mathadon** [comment]
@casella I'll have a look at it! The OMC errors seem quite self-explanatory so I'll try to get most of the problems out of the way based on the test report.

**Comment by casella** [comment]
> @casella I'll have a look at it! The OMC errors seem quite self-explanatory so I'll try to get most of the problems out of the way based on the test report.

OK. Given you merged your PR #1274, you should already see some improvement in tomorrow's report for the development version.

I guess most of the issues are due to technically illegal code:
- access to protected element: this is fixed by making it public, if you don't want to see it in the result file you can use the Hide annotation
- more missing 'each' in modifiers
- using stuff from partial packages, e.g. PartialMedium; the trick here is to use the redeclared medium instead, or alternatively to locally redeclare a non-partial package that extends the partial one and use it for the problematic declarations

There are a pair of issues that look like OMC bugs, I'll open tickets on the OpenModelica tracker and have @perost look at them.

I guess you should be able to get >98% success rate easily. The works we did to improve support of Buildings obviously paid off here 😄

**Comment by Mathadon** [comment]
Indeed, the most complex stuff from a compiler point of view (nested expandable or complex connectors in inner/outer models) seems to be supported since the main example models run fine.
